ActorSystem: build errors in C++23 mode

Apparently actor system has build errors when `"cxx_std": "c++23",` is specified in `build/ya.conf.json` for clang-18.

```
In file included from $(SOURCE_ROOT)/contrib/ydb/library/actors/core/executor_thread.cpp:1:
In file included from $(SOURCE_ROOT)/contrib/ydb/library/actors/core/executor_thread.h:3:
In file included from $(SOURCE_ROOT)/contrib/ydb/library/actors/core/defs.h:5:
In file included from $(SOURCE_ROOT)/contrib/ydb/library/actors/util/defs.h:6:
In file included from $(SOURCE_ROOT)/util/generic/bt_exception.h:4:
In file included from $(SOURCE_ROOT)/util/generic/yexception.h:5:
In file included from $(SOURCE_ROOT)/util/generic/string.h:3:
In file included from $(SOURCE_ROOT)/contrib/libs/cxxsupp/libcxx/include/algorithm:1793:
In file included from $(SOURCE_ROOT)/contrib/libs/cxxsupp/libcxx/include/__algorithm/inplace_merge.h:28:
$(SOURCE_ROOT)/contrib/libs/cxxsupp/libcxx/include/__memory/unique_ptr.h:67:19: error: invalid application of 'sizeof' to an incomplete type 'NActors::IEventHandle'
   67 |     static_assert(sizeof(_Tp) >= 0, "cannot delete an incomplete type");
      |                   ^~~~~~~~~~~
$(SOURCE_ROOT)/contrib/libs/cxxsupp/libcxx/include/__memory/unique_ptr.h:298:7: note: in instantiation of member function 'std::default_delete<NActors::IEventHandle>::operator()' requested here
  298 |       __ptr_.second()(__tmp);
      |       ^
$(SOURCE_ROOT)/contrib/libs/cxxsupp/libcxx/include/__memory/unique_ptr.h:264:71: note: in instantiation of member function 'std::unique_ptr<NActors::IEventHandle>::reset' requested here
  264 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
      |                                                                       ^
$(SOURCE_ROOT)/contrib/ydb/library/actors/core/event.h:32:37: note: in instantiation of member function 'std::unique_ptr<NActors::IEventHandle>::~unique_ptr' requested here
   32 |             return DoExecute(actor, std::move(eventPtr));
      |                                     ^
$(SOURCE_ROOT)/contrib/ydb/library/actors/core/event_load.h:10:11: note: forward declaration of 'NActors::IEventHandle'
   10 |     class IEventHandle;
      |           ^
1 error generated.
```

--- a/ydb/library/actors/core/event.cpp
+++ b/ydb/library/actors/core/event.cpp
@@ -50,4 +50,14 @@ namespace NActors {
         }
         return new TEventSerializedData;
     }
+
+    bool IEventBase::DoExecute(IActor* /*actor*/, std::unique_ptr<IEventHandle> /*eventPtr*/) {
+        Y_DEBUG_ABORT_UNLESS(false);
+        return false;
+    }
+
+    bool IEventBase::Execute(IActor* actor, std::unique_ptr<IEventHandle> eventPtr) {
+        return DoExecute(actor, std::move(eventPtr));
+    }
+
 }

--- a/ydb/library/actors/core/event.h
+++ b/ydb/library/actors/core/event.h
@@ -18,19 +18,15 @@ namespace NActors {
         : TNonCopyable {
     protected:
         // for compatibility with virtual actors
-        virtual bool DoExecute(IActor* /*actor*/, std::unique_ptr<IEventHandle> /*eventPtr*/) {
-            Y_DEBUG_ABORT_UNLESS(false);
-            return false;
-        }
+        virtual bool DoExecute(IActor* actor, std::unique_ptr<IEventHandle> eventPtr);
+
     public:
         // actual typing is performed by IEventHandle
 
         virtual ~IEventBase() {
         }
 
-        bool Execute(IActor* actor, std::unique_ptr<IEventHandle> eventPtr) {
-            return DoExecute(actor, std::move(eventPtr));
-        }
+        bool Execute(IActor* actor, std::unique_ptr<IEventHandle> eventPtr);
 
         virtual TString ToStringHeader() const = 0;
         virtual TString ToString() const {

